alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -109,6 +101,14 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
         val docsToQueries = mutableMapOf<String, MutableList<String>>()
 
         try {
+            monitorCtx.docLevelMonitorQueries!!.initDocLevelQueryIndex(monitor.dataSources)
```

Comment:
why did we shift this snippet?

Reply:
the method `monitorCtx.docLevelMonitorQueries!!.indexDocLevelQueries`  assumes that the `customer provided log index` always exist otherwise it throws an `Unhandled error`.

https://github.com/opensearch-project/alerting/blob/2ecde357581a968195d1a804c601361743084688/alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt#L105

So, moved it into the `try` block, so that we always get an `alert` with `ERROR` state for any `unhandled errors`.